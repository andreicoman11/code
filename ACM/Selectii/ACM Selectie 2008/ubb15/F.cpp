#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <map>
#include <list>
#include <cmath>
#include <algorithm>
#include <set>
using namespace std;

ifstream fin ("F.in");
//ofstream fout ("F.out");

int N;

int main()
{
	fin >> N;
	while(N--)
	{
		int n;
		fin >> n;
		int last;
		fin >> last;
		int len = 1;
		bool bigger = true;
		for(int i=1; i<n; i++)
		{
			int aux;
			fin >> aux;
			if( aux<last && bigger )
			{
				bigger = !bigger;
				len++;
			}
			if( aux>last && !bigger )
			{
				bigger = !bigger;
				len++;
			}
			last = aux;
		}
		cout << len << "\n";
	}

	return 0;
}