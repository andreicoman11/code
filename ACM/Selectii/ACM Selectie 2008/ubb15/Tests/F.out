1
2
5
3
2
668
1331
2035
2687
3302
3996
4683
5349
5987
6632
7344
8028
8646
9308
10118
10640
11386
12065
12743
13460
14106
14562
15297
16018
16677
17332
17949
18689
19367
20037
